### PR TITLE
Implement `getMetadata` for some more Passive scripts

### PR DESCRIPTION
**Pull request by ricekot**
Update the following scripts to implement the `getMetadata()` function:
- passive/google_api_keys_finder.js
- passive/JavaDisclosure.js
- passive/Report non static sites.js
- passive/RPO.js
- passive/s3.js

Part of #440.

### CODE REVIEW

File: passive/Report non static sites.js [hunk 1]
```diff
@@ -4,58 +4,61 @@
...
+function getMetadata() {
+  return ScanRuleMetadata.fromYaml(`
+id: 100017
+name: Non Static Site Detected
+description: >
+  A query string or form has been detected in the HTTP response body.
+  This indicates that this may not be a static site.
+solution: >
+  If this is not a static site then ignore or disable this rule.
+risk: high
```
Comment (kingthorin) [inline passive/Report non static sites.js:20]: I understand you've recreated what was previously here but ummm imho we are safe to change this

Comment (kingthorin) [reply passive/Report non static sites.js:20]: I guess if it is changed that should be noted in the changelog

Comment (ricekot) [reply passive/Report non static sites.js:20]: Should I change it to `info`?

Comment (kingthorin) [reply passive/Report non static sites.js:20]: That seems reasonable to me

File: passive/s3.js [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
-    "Remove s3 Buckets name from response or make sure the permissions in bucket are configured properly.";
-  var cweId = 200;
-  var wascId = 13;
+var ScanRuleMetadata = Java.type(
+  "org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata"
+);
 
+function getMetadata() {
+  return ScanRuleMetadata.fromYaml(`
+id: 100035
```
Comment (thc202) [inline passive/s3.js:9]: Same ID as `Information Disclosure - Java Stack Trace` (also in `scanners.md`).

File: passive/Report non static sites.js (hunk 1 above)
Comment (thc202) [reply passive/Report non static sites.js:20]: Regarding the changelog, some names/descriptions/… were changed as well (not only in this PR), maybe add a generic entry saying that the scripts that now implement `getMetadata()` had their info revised?

File: passive/s3.js (hunk 2 above)
Comment (ricekot) [reply passive/s3.js:9]: Wow, funny how we all missed that.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**